src/python/pants/engine/internals/graph_test.py

```diff
@@ -2233,6 +2243,34 @@ def test_depends_on_generated_targets(dependencies_rule_runner: RuleRunner) -> N
     )
 
 
+def test_plugin_transitive_excludes(dependencies_rule_runner: RuleRunner) -> None:
+    dependencies_rule_runner.write_files(
+        {
+            "transitive_dep.st": "",
+            "direct_dep.st": "//:transitive_dep.st",
+            "file.st": dedent(
+                """\
+                //:direct_dep.st
+                !!//:transitive_dep.st
```

Comment:
Without this line the test fails, as expected, because the transitive dep is returned in `TransitiveTargets`.